site/docs/patterns/wizard.mdx

```diff
@@ -0,0 +1,278 @@
+
+<Diagram
+  src="/img/patterns/wizard/wizard-validation-warning.png"
+  srcDark="/img/patterns/wizard/wizard-validation-warning-dark.png"
+  alt="Wizard stepper displaying a warning validation state after step validation"
+  border
+  background="secondary"
+  caption="The step in the stepper displays a warning state."
+/>
+
+### Error
+
+When a step contains an error after validation:
+- The step displays an error state
+- The application highlights the issues in the specific fields or areas that need attention to be resolved.
+- The next or submit button remains enabled for accessibility and discoverability, but clicking it again should refocus or re-highlight unresolved errors.
+- The cancel and previous buttons remain enabled, allowing users to exit or navigate backward in the flow.
+- If the validation passes, the flow proceeds to the next step or confirmation screen.
+
+<Diagram
+  src="/img/patterns/wizard/wizard-validation-error.png"
+  srcDark="/img/patterns/wizard/wizard-validation-error-dark.png"
+  alt="Wizard stepper displaying an error validation state"
+  border
+  background="secondary"
+  caption="The step in the stepper displays an error state"
+/>
+
+## Confirmation messages
+
+In certain use cases, you may want to provide additional feedback to users after an action is triggered. This includes showing warning, success, or other confirmation messages that help users stay informed and confident in their next steps.
+
+Confirmation messages can appear:
+- as a dialog over the app’s main content area, or
+- as part of the wizard flow within a modal wizard.
+
+### Warning message
+
+When the user clicks the cancel button, display a warning message if there’s a risk of losing progress. This is especially useful in lengthy or critical flows where saving is not supported. After confirming the cancel action, it should restore the state to how it was before the wizard began.
+
+#### Wizard with the warning dialog
+
+<Diagram
+  src="/img/patterns/wizard/wizard-confirmation-embedded-warning.png"
+  srcDark="/img/patterns/wizard/wizard-confirmation-embedded-warning-dark.png"
+  alt="Embedded warning dialog in wizard confirming cancel action and risk of losing progress"
+  border
+  background="secondary"
+  caption="The warning message shows up to confirm if cancelling."
+/>
+
+#### Modal wizard with the warning message
+
+<Diagram
+  src="/img/patterns/wizard/wizard-confirmation-modal-warning.png"
+  srcDark="/img/patterns/wizard/wizard-confirmation-modal-warning-dark.png"
+  alt="Modal wizard displaying warning message to confirm cancel action"
+  border
+  background="secondary"
+  caption="The warning message shows up to confirm if cancelling."
+/>
+
+## Completion message
+
+Use a completion message to confirm that the wizard task has been successfully completed. This helps users clearly understand that their action was successful and provides closure to the flow, especially when the task is critical. 
```

Comment:
We don't need to change this but what if the task wasn't successfully completed?